mlir/include/mlir/Analysis/Presburger/IntegerRelation.h

```diff
@@ -754,6 +751,30 @@ class IntegerRelation {
     return computeConstantLowerOrUpperBound<isLower>(pos).map(int64FromMPInt);
   }
 
+  /// Checks for emptiness by performing variable elimination on all
+  /// variables, running the GCD test on each equality constraint, and
+  /// checking for invalid constraints. Returns true if the GCD test fails for
+  /// any equality, or if any invalid constraints are discovered on any row.
+  /// Returns false otherwise.
+  bool isEmptyByFMTest() const;
```

Comment:
Can some of these tests conservatively return "false" or "true" if no certain decision could be made? I.e., false positives / negatives? If so, can we explicitly mention that for each of the function calls?